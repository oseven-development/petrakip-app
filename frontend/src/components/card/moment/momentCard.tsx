import { AudioVariantProps, AudioVariantCardBody } from './audioVariantCardBody'
import { PropsCardWrapper, CardWrapper } from './cardWrapper'
import { ImageVariantProps, ImageVariantCardBody } from './imageVariantCardBody'
import { TextVariantProps, TextVariantCardBody } from './textVariantCardBody'
import { VideoVariantProps, VideoVariantCardBody } from './videoVariantCardBody'

export type Variant = 'text' | 'image' | 'audio' | 'video'

export interface DefaultProps extends Omit<PropsCardWrapper, 'children'> {
  variant: Variant
}

export type TextVariantWithProps = TextVariantProps & DefaultProps
export type ImageVariantWithProps = ImageVariantProps & DefaultProps
export type AudioVariantWithProps = AudioVariantProps & DefaultProps
export type VideoVariantWithProps = VideoVariantProps & DefaultProps

type Props =
  | TextVariantWithProps
  | ImageVariantWithProps
  | AudioVariantWithProps
  | VideoVariantWithProps

function Card(props: Props): JSX.Element {
  const { title, subtitle, momentId } = props

  return (
    <CardWrapper {...{ title, subtitle, momentId }}>
      {props.variant === 'text' && <TextVariantCardBody {...props} />}
      {props.variant === 'image' && <ImageVariantCardBody {...props} />}
      {props.variant === 'audio' && <AudioVariantCardBody {...props} />}
      {props.variant === 'video' && <VideoVariantCardBody {...props} />}
    </CardWrapper>
  )
}

export { Card as MomentCard }
