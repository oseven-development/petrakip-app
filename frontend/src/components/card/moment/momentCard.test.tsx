import { render } from '@testing-library/react'

import { MomentCard } from './momentCard'

describe('MomentCard Component', () => {
  test.each`
    variant    | propType      | context
    ${'text'}  | ${'text'}     | ${'this is my text'}
    ${'image'} | ${'imageSrc'} | ${'https://imageSrc.de'}
    ${'video'} | ${'videoSrc'} | ${'https://videoSrc.de'}
    ${'audio'} | ${'audioSrc'} | ${'https://audioSrc.de'}
  `(
    '`when provide $variant variant the card should render a $variant card`',
    ({ variant, propType, context }) => {
      const title = ''
      const momentId = 123
      const { container } = render(
        //@ts-ignoreu
        <MomentCard {...{ title, momentId, variant, [propType]: context }} />,
      )
      expect(container).toMatchSnapshot()
    },
  )
})
