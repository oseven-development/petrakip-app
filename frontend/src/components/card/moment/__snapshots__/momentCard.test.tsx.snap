// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MomentCard Component \`when provide audio variant the card should render a audio card\` 1`] = `
<div>
  <ion-card>
    <ion-card-header>
      <ion-card-subtitle />
      <ion-router-link
        href="/moments/details/123"
        router-link="/moments/details/123"
      >
        <ion-card-title />
      </ion-router-link>
    </ion-card-header>
    <ion-card-content>
      <ion-grid>
        <ion-row
          class="ion-align-items-center"
        >
          <ion-col
            class="ion-text-center"
          >
            <ion-button
              fill="clear"
              size="large"
            >
              <ion-icon
                color="primary"
                icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Play Circle</title><path d='M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48zm74.77 217.3l-114.45 69.14a10.78 10.78 0 01-16.32-9.31V186.87a10.78 10.78 0 0116.32-9.31l114.45 69.14a10.89 10.89 0 010 18.6z'/></svg>"
                slot="icon-only"
              />
            </ion-button>
          </ion-col>
        </ion-row>
      </ion-grid>
    </ion-card-content>
  </ion-card>
</div>
`;

exports[`MomentCard Component \`when provide image variant the card should render a image card\` 1`] = `
<div>
  <ion-card>
    <ion-card-header>
      <ion-card-subtitle />
      <ion-router-link
        href="/moments/details/123"
        router-link="/moments/details/123"
      >
        <ion-card-title />
      </ion-router-link>
    </ion-card-header>
    <ion-card-content>
      <ion-img
        src="https://imageSrc.de"
      />
    </ion-card-content>
  </ion-card>
</div>
`;

exports[`MomentCard Component \`when provide text variant the card should render a text card\` 1`] = `
<div>
  <ion-card>
    <ion-card-header>
      <ion-card-subtitle />
      <ion-router-link
        href="/moments/details/123"
        router-link="/moments/details/123"
      >
        <ion-card-title />
      </ion-router-link>
    </ion-card-header>
    <ion-card-content>
      <ion-text>
        this is my text
      </ion-text>
    </ion-card-content>
  </ion-card>
</div>
`;

exports[`MomentCard Component \`when provide video variant the card should render a video card\` 1`] = `
<div>
  <ion-card>
    <ion-card-header>
      <ion-card-subtitle />
      <ion-router-link
        href="/moments/details/123"
        router-link="/moments/details/123"
      >
        <ion-card-title />
      </ion-router-link>
    </ion-card-header>
    <ion-card-content>
      <video
        controls=""
        width="100%"
      >
        <source
          src="https://videoSrc.de"
          type="video/mp4"
        />
        Your browser does not support the video tag.
      </video>
    </ion-card-content>
  </ion-card>
</div>
`;
